x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -783,40 +764,44 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
 
   const optionsData = useMemo(
     () =>
-      isEmpty(indexPattern.fields)
+      dataView == null || dataView?.fields == null || isEmpty(dataView?.fields)
```

Comment:
ideally this was never copy pasted from https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/timelines/components/timeline/query_bar/eql/index.tsx and was reused, and also should be a custom hook or something that doesn't iterate over fields 6 times, 

```
import { useMemo } from 'react';
import type { DataView } from '@kbn/data-views-plugin/common';
import type { EqlOptionsData } from '@kbn/timelines-plugin/common';

export const useEqlQueryOptions = (dataView: DataView) => {
  return useMemo(() => {
    const emptyOptions: EqlOptionsData = { keywordFields: [], dateFields: [], nonDateFields: [] };
    return Object.values(dataView.fields || {}).reduce((acc, field) => {
      if (field.esTypes?.includes('keyword')) {
        acc.keywordFields.push({ label: field.name });
      } else if (field.type === 'date') {
        acc.dateFields.push({ label: field.name });
      } else {
        acc.nonDateFields.push({ label: field.name });
      }
      return acc;
    }, emptyOptions);
  }, [dataView]);
};
```